yt/utilities/fortran_utils.py

```diff
@@ -136,16 +132,7 @@ def read_cattrs(f, attrs, endian="="):
         if n == 1:
             v = v[0]
         if isinstance(a, tuple):
-            if len(a) != len(v):
-                raise OSError(
-                    "An error occurred while reading a Fortran "
-                    "record. Record length is not equal to expected "
-                    "length: %s %s",
-                    len(a),
-                    len(v),
-                )
-
-            for k, val in zip(a, v):
+            for k, val in zip(a, v, strict=True):
```

Comment:
I do think the other version was somewhat better, as otherwise we're left deducing what caused the iterators to be of different lengths.

Reply:
FTR I'm waiting on @chrishavlin to weigh in but happy to change it if he agrees

Reply:
i think i prefer keeping the OSError, but don't feel too strongly about it.

Reply:
reverted !